tests/Integration/RequestIntegrationTest.php

```diff
@@ -31,20 +36,29 @@ public function testConstructorThrowsDeprecationIfPassingOptions(): void
      * @group legacy
      *
      * @dataProvider applyToEventWithRequestHavingIpAddressDataProvider
-     *
-     * @expectedDeprecation The "Sentry\Integration\RequestIntegration::applyToEvent" method is deprecated since version 2.1 and will be removed in 3.0.
      */
     public function testInvokeWithRequestHavingIpAddress(bool $shouldSendPii, array $expectedValue): void
     {
         $event = new Event();
         $event->getUserContext()->setData(['foo' => 'bar']);
 
         $request = new ServerRequest('GET', new Uri('http://www.example.com/fo'), [], null, '1.1', ['REMOTE_ADDR' => '127.0.0.1']);
-        $integration = new RequestIntegration(new Options(['send_default_pii' => $shouldSendPii]));
+        $integration = new RequestIntegration(new Options(['send_default_pii' => $shouldSendPii]), $this->createRequestFetcher($request));
+        $integration->setupOnce();
 
-        RequestIntegration::applyToEvent($integration, $event, $request);
+        /** @var ClientInterface&MockObject $client */
+        $client = $this->createMock(ClientInterface::class);
+        $client->expects($this->once())
+            ->method('getIntegration')
+            ->willReturn($integration);
 
-        $this->assertEquals($expectedValue, $event->getUserContext()->toArray());
+        SentrySdk::getCurrentHub()->bindClient($client);
+
+        withScope(function (Scope $scope) use ($event, $expectedValue): void {
+            $event = $scope->applyToEvent($event, []);
+            $this->assertNotNull($event);
```

Comment:
Since you're there, can you please add a blank line above this one to improve readability?